File: oop_6/cs/Letters.cs
namespace oop_6_cs
{
    class Letters : Strings
    {
        private char[] letterline;
        public Letters(char[] line)
        {
            letterline = line;
        }
        public override char[] GetLine()
        {
            return letterline;
        }
        public override int Length()
        {
            return letterline.Length;
        }
        public override void DeleteSymbol(char symbol = 'a')
        {
            int counterSymbols = 0;
            for (int i = 0; i < letterline.Length; i++)
            {
                if (letterline[i] == 'a')
                {
                    counterSymbols++;
                }
            }
            char[] result = new char[letterline.Length - counterSymbols];
            int j = 0;
            for (int i = 0; i < letterline.Length; i++)
            {
                if (letterline[i] != 'a')
                {
                    result[j] = letterline[i];
                    j++;
                }
            }
            letterline = result;
        }
    }
}

File: oop_6/cpp/Source Files/Letters.cpp
#include "Letters.h"

Letters::Letters(char* line)
{
    letterline = line;
}
Letters::~Letters(){}

char* Letters::GetLine()
{
    return letterline;
}
int Letters::Length()
{
    int length = 0;
    int i = 0;
    while (letterline[i] != '\0')
    {
        length++;
        i++;
    }
    return length;
}
void Letters::DeleteSymbol(char symbol = 'a')
{
    int counterSymbols = 0;
    for (int i = 0; i < Length(); i++)
    {
        if (letterline[i] == 'a')
        {
            counterSymbols++;
        }
    }
    char* result = new char[Length() - counterSymbols];
    int j = 0;
    for (int i = 0; i < Length(); i++)
    {
        if (letterline[i] != 'a')
        {
            result[j] = letterline[i];
            j++;
        }
    }
    result[j] = '\0';
    delete[] letterline;
    letterline = result;
}
